integration/spark/src/main/common/java/io/openlineage/spark/api/AbstractQueryPlanInputDatasetBuilder.java

```diff
@@ -0,0 +1,22 @@
+package io.openlineage.spark.api;
+
+import io.openlineage.client.OpenLineage;
+import org.apache.spark.scheduler.SparkListenerEvent;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionStart;
+
+public abstract class AbstractQueryPlanInputDatasetBuilder<P extends LogicalPlan>
```

Comment:
I'm not really sure I understand the reasoning behind triggering this builder only on start events. Please add javadoc here explaining the intended use of this class and try to make it clear to users that only start events will trigger subclasses

Reply:
The behavior is repeated for all Spark 3 visitors so that it gets triggered only for start/stop events. Javadoc added.